### PR TITLE
Add review like and dislike interactions

### PR DESCRIPTION
**Pull request by terror**

### CODE REVIEW

File: client/src/components/CourseReview.tsx [hunk 1]
```diff
@@ -132,9 +217,34 @@ export const CourseReview = ({
...
+          </div>
+          <button className='flex h-8 w-8 items-center justify-center rounded-md text-gray-700 focus:outline-none dark:text-white'>
+            <AiFillDislike
+              onClick={handleDislike}
+              className={classNames(
+                'h-4 w-4',
+                kind === InteractionKind.Dislike ? 'fill-red-600' : ''
+              )}
+            />
+          </button>
```
Comment (JeffDotPng) [inline client/src/components/CourseReview.tsx:246]: Would probably be cleaner to extract this into its own component along with the interaction logic

File: client/src/components/CourseReview.tsx [hunk 2]
```diff
@@ -27,22 +31,103 @@ export const CourseReview = ({
...
+              (interaction) => interaction.kind === InteractionKind.Dislike
+            ).length
+        );
+        if (user)
+          setKind(
+            interactions.find(
+              (interaction: Interaction) => interaction.referrer === user.id
+            )?.kind
+          );
+      })
```
Comment (JeffDotPng) [inline client/src/components/CourseReview.tsx:72]: Do you think it'd be better to do this data processing in the backend? Thinking about potential performance problems in the future

File: crates/db/src/db.rs [hunk 3]
```diff
@@ -202,6 +203,97 @@ impl Db {
...
+          .await?;
+      }
+      None => {
+        self
+          .database
+          .collection::<Interaction>(Self::INTERACTION_COLLECTION)
+          .insert_one(interaction, None)
+          .await?;
+      }
+    }
```
[the remark this answers is not in the document]
Comment (terror) [reply crates/db/src/db.rs]: Nice catch, upsert is clean

File: client/src/components/CourseReview.tsx (hunk 2 above)
Comment (terror) [reply client/src/components/CourseReview.tsx]: Yah good call

**Pull request merged**